### DISCUSSION TITLE
Looking for End of support date for System.compostion packages

### DISCUSSION THREAD

**Discussion by Aishbeck** [dotnet/runtime#108532] [General] [answered]
I am looking for end of support / life details for the following packages.
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/asekar/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/asekar/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#467886" vlink="#96607D">

system.composition.attributedmodel | 4.6.24705.01
-- | --
system.composition.convention | 4.6.24705.01
system.composition.hosting | 4.6.24705.01
system.composition.runtime | 4.6.24705.01
system.composition.typedparts | 4.6.24705.01

</body>

</html>

**Comment by huoyaoyuan** [comment]
There's no such package versions. Their package versions bumped from 1.4.1 to 5.0.0, and continuously provided for 9.0.0.
The version scheme 4.6.24705.01 looks like assembly version for .NET Framework.

**Comment by elgonzo** [comment]
The package version as listed in the respective DLLs as assembly version. 4.6.24705.01 is the product/file version correlating with package/assembly version **1.0.31**, as far as System.Composition.AttributedModel is concerned. (I blindly assume this also applies to the other System.Composition.* packages as well)

Here the metadata (and assorted info) of the System.Composition.AttributedModel.dll from the 1.0.31 package:

| Property | Value |
| --- | --- |
| Product Version | 0.0.0.0 | 
| File Version | 4.6.24705.1 | 
| File Type | DLL | 
| File Subtype | "" | 
| File Flags | "" | 
| File OS | "32-bit Windows" | 
| Comments | System.Composition.AttributedModel | 
| Company Name | "Microsoft Corporation" | 
| File Description | System.Composition.AttributedModel | 
| File Version | 4.6.24705.01 | 
| Internal Name | System.Composition.AttributedModel.dll | 
| Legal Copyright | "© Microsoft Corporation.  All rights reserved." | 
| Original File Name | System.Composition.AttributedModel.dll | 
| Product Name | "Microsoft® .NET Framework" | 
| Product Version | "4.6.24705.01. Commit Hash: 4d1af962ca0fede10beb01d197367c2f90e92c97" | 
| Assembly Version | 1.0.31.0 | 
| Language | "Language Neutral (0)" | 
| Code Page | 1200 | 
| File Created | "Fri 04.10.2024 16:14:00" | 
| File Modified | "Sat 05.11.2016 04:55:42" | 
| File Accessed | "Fri 04.10.2024 16:14:00" | 
| File Size | "24,840 bytes" | 
| MD5 | 24a0c8cce8c132df82c9b9c1ae834d05 | 
| SHA1 | 1ce7555a43e74f5ac95fc734505068c4e0c4a78e | 
| "Characteristics" | "0x2022 (Executable, Large Address Aware, DLL)" | 
| "Dll Characteristics" | "0x8540 (Dynamic Base, NX Compatible, No SEH, Terminal Server Aware)" | 
| "CLR Header" (.NET) | ✓ | 
| Entry Point Raw Offset | 0x000019D6 | 
| Compiler (PE Sniffer) | "Microsoft Visual C# / Basic .NET" | 
| Compiler (PEID) | "Microsoft Visual C# / Basic .NET" |

**Reply by Aishbeck** [reply]
Is there a end of date / year for support that I can get for documentation for his version?

**Reply by huoyaoyuan** [reply, marked as answer]
It's the very first version released with .NET Core and should have reached EOL at 2019. The latest versions of the package is supported.

**Reply by Aishbeck** [reply]
Thanks, Is it the same year for system.io.packaging 3.1.0 ?

**Reply by huoyaoyuan** [reply]
I'm never aware of such version in any field of any version of System.IO.Packaging. The first version of the package is 4.0.0 with assembly version 4.x. The only potentially related version is package version 4.7.0, which released together with .NET Core 3.1 and has AssemblyInformationalVersion set to 3.1.0-commithash.

**Reply by Aishbeck** [reply]
Is it possible that it could be the assembly version 3.1.0?

**Reply by huoyaoyuan** [reply]
No, the first assembly version was 4.0.0.0, and the file version was 1.0.x.

**Reply by Aishbeck** [reply]
okay typically when does support ends for a package specific version? is it till the next version is released or is it supported even though there is a higher version of it , example System.Composition.AttributedModel  has latest version 9.0.0 which is supported, what is the support ccle for other previous versions?

**Reply by Aishbeck** [reply]
I want to know this support details for all Microsoft packages

**Reply by huoyaoyuan** [reply]
The full support policy can be complex and I can't speak for it precisely. For "mainline" packages with versions aligned to current .NET versions, you can expect the same lifetime of the same .NET version. For example, System.Composition.AttributedModel 8.0.x will be supported as long as .NET 8 is supported.
